### PR TITLE
Add "full" example

### PR DESCRIPTION
**Pull request by tp**
This shouldn't be our only example, just wanted to create a fuller example.

It's still WIP as I want to see how we could split out the UI and the actual login.

Also, even though nothing bad would happen if not implemented, I wanted to see how to add some "disable button immediately" functionality would look. Not sure yet if the below approach is even guaranteed in Flutter to guard against that in all circumstances. For all practical purposes it's probably sufficient to just do it based on the bloc's state.

### CODE REVIEW

File: packages/with_bloc/example/main.dart [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    });
+
+    return completer.future;
+  }
+}
+
+class App extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    WithBloc<LoginBloc, LoginState>(
```
Comment (HenriBeck) [inline packages/with_bloc/example/main.dart]: Shouldn't we `return` here?

File: packages/with_bloc/example/main.dart [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+          obscureText: true,
+          decoration: InputDecoration(
+            border: OutlineInputBorder(),
+            labelText: 'Username',
+          ),
+        ),
+        MaterialButton(
+          child: Text('Login'),
+          onPressed: !_loginInProgress
+              ? () async {
```
Comment (HenriBeck) [inline packages/with_bloc/example/main.dart]: Just for proper measures, I would make this a class method not to have inline methods inside the build function

**Comment by HenriBeck** [comment]
@tp Do you know if the example needs to be a full flutter app? I thought that's what's being generated when you do `flutter create package`

**Comment by tp** [comment]
> Do you know if the example needs to be a full flutter app?

No it can just be a sample file, not sure if it even gets compiled. 

But I guess having a proper, startle Flutter app can make sense, in which case one is supposed to add a `README.md` to the example directory and link to the individual examples oneself: https://dart.dev/tools/pub/package-layout#examples

File: packages/with_bloc/example/main.dart (hunk 1 above)
Comment (tp) [reply packages/with_bloc/example/main.dart]: Absolutely!

File: packages/with_bloc/example/main.dart (hunk 2 above)
Comment (tp) [reply packages/with_bloc/example/main.dart]: Done.

**Pull request merged**